examples/pytorch/graphsage_base/node_classification.py

```diff
@@ -0,0 +1,141 @@
+        with torch.no_grad():
+            x = blocks[0].srcdata['feat']
+            ys.append(blocks[-1].dstdata['label'])
+            y_hats.append(model(blocks, x))
+    return MF.accuracy(torch.cat(y_hats), torch.cat(ys))
+
+def layerwise_infer(args, device, graph, nid, model, batch_size):
+    model.eval()
+    nid = nid.to(device)
+    with torch.no_grad():
+        pred = model.inference(graph, device, batch_size)
+        pred = pred[nid].to(device)
+        label = graph.ndata['label'][nid].to(device)
+        return MF.accuracy(pred, label)
+
+def train(args, device, g, dataset, model):
+    # create sampler & dataloader
+    train_idx = dataset.train_idx.to(device)
+    val_idx = dataset.val_idx.to(device)
+    sampler = dgl.dataloading.NeighborSampler([15, 10, 5],
+                                              prefetch_node_feats=['feat'],
+                                              prefetch_labels=['label'])
+    use_uva = (args.mode == 'mixed')
+    train_dataloader = dgl.dataloading.DataLoader(g, train_idx, sampler, device=device,
+                                                  batch_size=1024, shuffle=True,
+                                                  drop_last=False, num_workers=0,
+                                                  use_uva=use_uva)
+
+    valid_dataloader = dgl.dataloading.DataLoader(g, val_idx, sampler, device=device,
+                                                  batch_size=1024, shuffle=True,
+                                                  drop_last=False, num_workers=0,
+                                                  use_uva=use_uva)
+
+    opt = torch.optim.Adam(model.parameters(), lr=1e-3, weight_decay=5e-4)
+    
+    for epoch in range(10):
+        model.train()
+        total_loss = torch.tensor(0.0).to(device)
+        for it, (input_nodes, output_nodes, blocks) in enumerate(train_dataloader):
+            x = blocks[0].srcdata['feat']
+            y = blocks[-1].dstdata['label']
+            y_hat = model(blocks, x)
+            loss = F.cross_entropy(y_hat, y)
+            opt.zero_grad()
+            loss.backward()
+            opt.step()
+            total_loss += loss
+        acc = evaluate(model, g, valid_dataloader)
+        mem = torch.cuda.max_memory_allocated() / 1000000
+        print("Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} | "
+              "GPU Mem(MB) {:.2f}".format(epoch, total_loss.item() / (it+1),
+                                          acc.item(), mem))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--mode", default='mixed', choices=['mixed', 'puregpu'],
+                        help="Training mode. 'mixed' for CPU-GPU mixed training. "
+                             "'puregpu' for pure-GPU training.")
+    args = parser.parse_args()
+    print(f'Training in {args.mode} mode.')
+
+    # load and preprocess dataset
+    dataset = dgl.data.AsNodePredDataset(DglNodePropPredDataset('ogbn-products'))

```

Comment:
Personally I prefer replacing `dgl.data.AsNodePredDataset` with `from dgl.data import AsNodePredDataset` and `dgl.dataloading.DataLoader` with `from dgl.dataloading import DataLoader`, though this might be highly dependent on personal taste.